reference-implementation/to-upstream-wpts/readable-byte-stream/readable-byte-stream.js

```diff
@@ -0,0 +1,1960 @@
+'use strict';
+
+if (self.importScripts) {
+  self.importScripts('../resources/rs-utils.js');
+  self.importScripts('/resources/testharness.js');
+}
+
+test(() => {
+  assert_throws(new TypeError(), () => new ReadableStream().getReader({ mode: 'byob' }));
+}, 'getReader({mode: "byob"}) throws on non-bytes streams');
+
+
+test(() => {
+  // Constructing ReadableStream with an empty underlying byte source object as parameter shouldn't throw.
+  new ReadableStream({ type: 'bytes' });
+}, 'ReadableStream with byte source can be constructed with no errors');
+
+async_test(t => {
+  let startCalled = false;
+  let controller;
+
+  new ReadableStream({
```

Comment:
Because of how the async_test framework works, I believe you need to wrap any functions called async that could possibly throw, such as `pull`, in `t.step_func`